Model/abstractShip.php

```diff
@@ -13,9 +13,13 @@ abstract class AbstractShip
     private $strength = 0;
 
     abstract public function getJediFactor(); // forces extended class to have this method
+
+    abstract public function setJediFactor($jediPower);
```

Comment:
I think you are adding the setters for `setJediFactor` and `setType` so that you can call them from the new/edit pages.

I think since you have 2 separate classes `Ship` and `RebelShip`, you don't need the setters and you can just keep the original getters as they were.

And you don't need to set the type, because it is set by the class itself.

Let me know if you want to discuss this further.